model/postgres/postgres_timeline.rb

```diff
@@ -53,7 +53,27 @@ def backups
 
   def last_backup_label_before_target(target:)
     backup = backups.sort_by(&:last_modified).reverse.find { _1.last_modified < target }
-    backup.key.delete_prefix("basebackups_005/").delete_suffix("_backup_stop_sentinel.json")
+    backup.key.delete_prefix("basebackups_005/").delete_suffix("_backup_stop_sentinel.json") if backup
+  end
+
+  def refresh_earliest_backup_time
+    update(earliest_backup_at: backups.map(&:last_modified).min)
+  end
+
+  # The "earliest_backup_at" column is used to cache the earliest backup time,
+  # eliminating the need to query the blob storage every time. The
+  # "earliest_backup_at" value can be changed when a new backup is created or an
+  # existing backup is deleted. Currently, we lack a backup cleanup feature and
+  # cannot determine whether a backup operation has completed. For the time
+  # being, we are using the "created_at" attribute if it's nil. Once the above
+  # features are implemented, we can discard the "created_at" attribute and
+  # invoke the "refresh_earliest_backup_time" method at the appropriate points.
+  def earliest_restore_time
+    (earliest_backup_at || created_at) + 5 * 60
```

Comment:
Actually, since you already implemented refresh_earliest_backup_time, why not just;

```ruby
def earliest_restore_time
    (earliest_backup_at || refresh_earliest_backup_time) + 5
end
```

Reply:
Makes sense. Done.